WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadActionUseCase.kt

```diff
@@ -0,0 +1,77 @@
+package org.wordpress.android.ui.uploads
+
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.UploadStore
+import org.wordpress.android.ui.posts.PostUtilsWrapper
+import org.wordpress.android.ui.uploads.UploadActionUseCase.UploadAction.DO_NOTHING
+import org.wordpress.android.ui.uploads.UploadActionUseCase.UploadAction.REMOTE_AUTO_SAVE
+import org.wordpress.android.ui.uploads.UploadActionUseCase.UploadAction.UPLOAD
+import org.wordpress.android.ui.uploads.UploadActionUseCase.UploadAction.UPLOAD_AS_DRAFT
+import org.wordpress.android.util.DateTimeUtils
+import java.util.Date
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val MAXIMUM_AUTO_UPLOAD_RETRIES = 10
+private const val TWO_DAYS_IN_MILLIS = 1000 * 60 * 60 * 24 * 2
+
+@Singleton
+class UploadActionUseCase @Inject constructor(
```

Comment:
Are you planning on adding tests for this class or do you think it's sufficiently covered by the `UploadStarterTest`?

Reply:
I'll definitely add tests for this class. Tbh I'll probably delete the tests in the UploadStarter and use a mock. Wdyt @shiki ?

Reply:
@shiki That's fine by me.